api/jobs.yaml

```diff
@@ -149,6 +149,19 @@ definitions:
         type: string
       display:
         type: string
+      editable:
+        type: boolean
+        default: false
+      fieldType:
+        type: string
+        default: text
+        enum:
```

Comment:
nit: I might separate this out into its own enum definition like we do for [Status](https://github.com/DataBiosphere/job-manager/blob/master/api/jobs.yaml#L350). This would allow `FileType` to be re-used if we wanted to also add it to say `CommonLabels`.